pkg/assertions/yamlmeta.go

```diff
@@ -3,25 +3,35 @@
 
 package assertions
 
-import "github.com/vmware-tanzu/carvel-ytt/pkg/yamlmeta"
+import (
+	"fmt"
+
+	"github.com/vmware-tanzu/carvel-ytt/pkg/yamlmeta"
+)
+
+const validations = "validations"
 
 // AddValidations appends validation Rules to node's validations metadata, later retrieved via GetValidations().
 func AddValidations(node yamlmeta.Node, rules []Rule) {
-	metas := node.GetMeta("validations")
-	if currRules, ok := metas.([]Rule); ok {
+	metas := node.GetMeta(validations)
+	if metas != nil {
+		currRules, ok := metas.([]Rule)
+		if !ok {
+			panic(fmt.Sprintf("Unexpected validations in meta : %s %v", node.GetPosition().AsCompactString(), metas))
+		}
 		rules = append(currRules, rules...)
 	}
 	SetValidations(node, rules)
 }
 
 // SetValidations attaches validation Rules to node's metadata, later retrieved via GetValidations().
 func SetValidations(node yamlmeta.Node, rules []Rule) {
-	node.SetMeta("validations", rules)
+	node.SetMeta(validations, rules)
 }
 
 // GetValidations retrieves validation Rules from node metadata, set previously via SetValidations().
 func GetValidations(node yamlmeta.Node) []Rule {
-	metas := node.GetMeta("validations")
+	metas := node.GetMeta(validations)
 	if rules, ok := metas.([]Rule); ok {
```

Comment:
🤔  wondering if we should not rely on a bad cast and panic if `validations` was set with something other than `[]Rule`?

Reply:
Discussion: Return the direct cast and let go take care of the panic. cast the value without checking.